draft-ietf-quic-transport.md

```diff
@@ -1427,6 +1431,63 @@ a single packet.
 
 In TLS, the Retry packet type is used to carry the HelloRetryRequest message.
 
+## ECN Capability Check {#ecn-capability-check}
+
+Explicit Congestion Notification (ECN) {{!RFC3168}} is feature that allows for
+non-destructive congestion notification by a network node.  That is, packets are
+marked instead of being discarded by routers and other devices along a network
+path. QUIC endpoints determine whether a path correctly supports ECN marking by
+verifying paths when connections are established and when migrating the
+connection to a new path. Each peer independently validates network paths, which
+leads to ECN being enabled separately for each direction on a path.
+
+ECN is validated by setting the ECT(0) or ECT(1) bit in the IP header, following
+the guidelines of {{!RFC8311}}. Clients and servers both mark the IP packet
+containing an Initial packet ({{packet-initial}}). If ECN is supported, the
+recipient of a marked packet will increase the corresponding counter and send an
+ACK_ECN frame.
+
+An endpoint uses the ACK_ECN frame to verify that the ECT markings were received
+by its peer. The endpoint counts the number of acknowledged QUIC packets that
+were sent in marked IP packets. If the counters in the ACK_ECN frame match that
+number, the path and peer do not remove ECN markings. The endpoint records this
+path as being ECN capable and continues to mark IP packets that it sends.
+
+If an endpoint receives an ACK frame, indicating that no markings were received,
+or the counter in the ACK_ECN frame do not match the number of QUIC packets that
+were sent in marked IP packets, then the path or peer remove ECN markings. The
+endpoint records this path as not being ECN capable and it ceases marking of
+packets.
+
+IP packets sent on a new path SHOULD be marked with ECT(0) or ECT(1) to verify
+that the new path supports ECN, see {{ecn-connection-migration}}.
+
+### Continuous Verification of ECN {#ecn-continuous-verification}
+
+If the ECN capability check was successful and the endpoint continues to send
+ECT marked packets then continuous verification is applied. This is to detect
+any cases when ECN field is bleached, that is, zeroed out by a network node,
```

Comment:
Above, the text used the term "remove markings". We should pick one term - zeroed, bleached, removed - consistently.